### PR TITLE
Refs [TASK][36315] : Click item Coin & Exchange

### PR DESCRIPTION
**Pull request by hideoncalm**
## Related Tickets
- [#Task 36315: Click item Coin & Exchange](https://edu-redmine.sun-asterisk.vn/issues/36315)
- [#Task 36317: Create Detail Coin Screen](https://edu-redmine.sun-asterisk.vn/issues/36317)
## WHAT
- 
## Evidence (Screenshot or Video)
image
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/coinqapp/ui/exchangedetail/ExchangeDetailActivity.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    companion object {
+        const val EXCHANGE = "EXCHANGE"
+
+        fun getInstance(context: Context, exchange: Exchange) =
+            Intent(context, ExchangeDetailActivity::class.java).apply {
+                putExtra(EXCHANGE, exchange)
+            }
+    }
```
Comment (trangnq258) [inline app/src/main/java/com/sunasterisk/coinqapp/ui/exchangedetail/ExchangeDetailActivity.kt]: xem lại convention về đặt tên EXTRA nhé
https://github.com/framgia/coding-standards/blob/master/eng/android/codingstyleguide.md
2.10

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -12,5 +12,6 @@
 
     <dimen name="sp_10">10sp</dimen>
     <dimen name="sp_15">15sp</dimen>
+    <dimen name="sp_20">20sp</dimen>
     <dimen name="sp_40">40sp</dimen>
```
Comment (trangnq258) [inline app/src/main/res/values/dimens.xml:16]: sắp xếp lại.

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
+            android:name=".ui.MainActivity"
+            android:theme="@style/Theme.CoinQ.NoActionBar" />
+        <activity
+            android:name=".ui.coindetail.CoinDetailActivity"
+            android:theme="@style/Theme.CoinQ.NoActionBar" />
+        <activity
+            android:name=".ui.exchangedetail.ExchangeDetailActivity"
+            android:theme="@style/Theme.CoinQ.NoActionBar" />
+        <activity
+            android:name=".ui.splash.SplashScreen"
```
Comment (thonc-2135) [inline app/src/main/AndroidManifest.xml]: sao dùng nhiều activity thế mà k dùng fragment nhỉ ??

File: app/src/main/java/com/sunasterisk/coinqapp/data/model/Exchange.kt [hunk 4]
```diff
@@ -6,14 +6,15 @@ import com.sunasterisk.coinqapp.utils.ExchangeConstant.EXCHANGE_NAME
...
+import java.io.Serializable
 
 data class Exchange(
     val id: String,
     val name: String,
     val image: String,
     val volume: Long,
     val trustScore: Int
-) {
+) : Serializable {
```
Comment (trangnq258) [inline app/src/main/java/com/sunasterisk/coinqapp/data/model/Exchange.kt]: sao k dùng Parcelable mà dùng Serializable nhỉ

File: app/src/main/java/com/sunasterisk/coinqapp/ui/coindetail/CoinDetailActivity.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        with(binding) {
+            buttonBack.setOnClickListener {
+                onBackPressed()
+                finish()
+            }
+        }
+    }
+
+    companion object {
+        const val EXTRA_COIN = "COIN"
```
Comment (trangnq258) [inline app/src/main/java/com/sunasterisk/coinqapp/ui/coindetail/CoinDetailActivity.kt]:
```suggestion
        const val EXTRA_COIN = "EXTRA_COIN"
```

File: app/src/main/java/com/sunasterisk/coinqapp/data/model/Exchange.kt (hunk 4 above)
Comment (hideoncalm) [reply app/src/main/java/com/sunasterisk/coinqapp/data/model/Exchange.kt]: tôi chưa tìm hiểu về Parcelable. để tôi sửa lại

File: app/src/main/java/com/sunasterisk/coinqapp/ui/coindetail/CoinDetailFragment.kt [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+            coin?.apply {
+                imageCoin.loadImage(image)
+                textCoinName.text = name
+                textCoinSymbol.text = symbol
+            }
+        }
+    }
+
+    companion object {
+        private const val BUNDLE_COIN = "COIN"
```
Comment (trangnq258) [inline app/src/main/java/com/sunasterisk/coinqapp/ui/coindetail/CoinDetailFragment.kt]:
```suggestion
        private const val BUNDLE_COIN = "BUNDLE_COIN"
```

File: app/src/main/java/com/sunasterisk/coinqapp/ui/exchangedetail/ExchangeDetailFragment.kt [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import com.sunasterisk.coinqapp.utils.backPress
+import com.sunasterisk.coinqapp.utils.loadImage
+
+class ExchangeDetailFragment : BaseFragment<FragmentExchangeDetailBinding>() {
+
+    private var exchange: Exchange? = null
+
+    override val binding by lazy { FragmentExchangeDetailBinding.inflate(layoutInflater) }
+
+
```
Comment (trangnq258) [inline app/src/main/java/com/sunasterisk/coinqapp/ui/exchangedetail/ExchangeDetailFragment.kt]: hai dòng trống

**Review by huytq-2422 — APPROVED** [review]

**Pull request merged**